### ISSUE TITLE
Should be able to pass additional FDs

### ISSUE THREAD

**Issue by akamensky** [sevlyar/go-daemon#53]
Right now there is not way to get additional FDs to be passed from parent to daemon.

My goal is to open TCP socket, then get FD (using [TCPListener.File()](https://golang.org/pkg/net/#TCPListener.File)), then pass that FD to the daemon and convert it back to Listener (using [net.FileListener()](https://golang.org/pkg/net/#FileListener))

I think this should be totally doable and not hard to implement, so I don't see why it was not there from the start. There are more uses to this functionality than just this.

**Comment by sherlockblaze** [comment]
@akamensky Hey man, did you get a implement now? I want the feature you mentioned too

**Comment by akamensky** [comment]
@sherlockblaze Nope, to get that working I would need to rewrite large part of this package.

**Comment by sherlockblaze** [comment]
@akamensky I think I got it already.
When you use the lib to start a daemon thread, you should set Args for it.
```go
cntxt := &daemon.Context{
		PidFileName: "pid",
		PidFilePerm: 0644,
		LogFileName: "log",
		LogFilePerm: 0640,
		WorkDir:     "./",
		Umask:       027,
		Args:        []string{"[go-daemon sample]", "test"},
	}
```
And the Args just like the command line you use. Like `nginx start`, arg[0] is ngxin and arg[1] is start and like. So, you just need to set the args with your command by using a golang slice, you can achieve your goal.

**Comment by akamensky** [comment]
@sherlockblaze command line arguments have nothing to do with file descriptors...

**Comment by sherlockblaze** [comment]
@akamensky sorry man, I misunderstood your purpose
